tests/test_layout.py

```diff
@@ -134,3 +134,66 @@ def test_layout_command(hlwm, running_clients, running_clients_num, num_splits):
     lines = [re.sub(r' [0-9]+% selection=[01]', '', l) for l in lines]
     assert dumped == '\n'.join(lines).replace(':', '')
 
+
+@pytest.mark.parametrize("running_clients_num,client2focus",
+            [(cnt, idx) for cnt in [3, 5] for idx in range(0, cnt)])
```

Comment:
Maybe I'm missing something here, but wouldn't it be simpler to do this with two `@pytest.mark.parametrize` decorators? It's a cartesian product of [3,5] and range(0, cnt), right?

Reply:
It's not a cartesian product, because `cnt` is not fixed by the first component, i.e. an element of `[3,5]`. Can I let one `@pytest.mark.parametrize` depend on another `@pytest.mark.parametrize`?

Reply:
The answer is: no. You can have dependencies between fixtures, but not between parameters. The following code fails:
```
@pytest.mark.parametrize("running_clients_num", [3, 5])
@pytest.mark.parametrize("client2focus", range(0, running_clients_num))
@pytest.mark.parametrize("num_splits", [0, 1, 2, 3])
def test_jumpto_within_tag(hlwm, running_clients, client2focus, num_splits):
    assert 0 == 0
```
It fails with:
```
_______________________ ERROR collecting tests/test_layout.py _______________________
tests/test_layout.py:139: in <module>
    @pytest.mark.parametrize("client2focus", range(0, running_clients_num))
E   NameError: name 'running_clients_num' is not defined
!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!
```

Reply:
Oh well, seems like i *was* missing something. ;-) Then I guess it's fine.